File: Go/safadao.go
package main

import "fmt"

func somatorio(mes int) int {
	soma := 0
	for i := mes; i > 0; i-- {
		soma += i
	}
	return soma
}
func wesley_safadao(dia, mes, ano int) {
	safadeza := float32(somatorio(mes)) + (float32(ano) / 100.0) * (50.0 - float32(dia))
	anjo := 100.0 - safadeza
	
	fmt.Printf("%.1f %% ANJO, mas o %.1f %% é VAGABUNDO!\n", anjo, safadeza)
}
func main() {
	var dia, mes, ano int
	
	fmt.Println("Insira sua data de nascimento: ");
	fmt.Print("Dia (dd): ");

	fmt.Scanf("%d", &dia);
	fmt.Print("Mes (mm): ");

	fmt.Scanf("%d", &mes);
	fmt.Print("Ano (AA): ");
	
	fmt.Scanf("%d", &ano);
	
	wesley_safadao(dia, mes, ano)
}

File: C/safadao.c
#include <stdio.h>

int somatorio(int mes) {
    int i, soma = 0;

    for (i = 1; i <= mes; i++) {
        soma += i;
    }

    return soma;
}

void wesley_safadao(int dia, int mes, int ano) {
    float safadeza, anjo;

    safadeza = somatorio(mes) + (ano/100) * (50-dia);
    anjo = 100 - safadeza;

    printf("%.1f %% ANJO, mas o %.1f %% eh VAGABUNDO!\n", anjo, safadeza);
}

int main(int argc, char const *argv[]) {
    int dia = 0, mes = 0, ano = 0;

    printf("Insira sua data de nascimento:\n");
    printf("Dia (dd): ");
    fflush(stdin);
    scanf("%d", &dia);
    printf("Mes (mm): ");
    fflush(stdin);
    scanf("%d", &mes);
    printf("Ano (AA): ");
    fflush(stdin);
    scanf("%d", &ano);

    wesley_safadao(dia, mes, ano);

    return 0;
}
